src/scanners/base.js

```diff
@@ -41,56 +41,38 @@ export default class BaseScanner {
     ensureFilenameExists(this.filename);
   }
 
-  scan(_rules = this._defaultRules) {
-    return new Promise((resolve, reject) => {
-      this.getContents()
-        .then((contents) => {
-          const promises = [];
-          // Ignore private functions exported in rule files.
-          //
-          // (These are exported for testing purposes, but we don't want
-          // to include them in our linter's rules.)
-          const rules = ignorePrivateFunctions(_rules);
-
-          Object.keys(rules).forEach((rule) => {
-            this._rulesProcessed++;
-            promises.push(rules[rule](contents, this.filename, this.options));
-          });
-
-          return Promise.all(promises);
-        })
-        .then((ruleResults) => {
-          ruleResults.forEach((messages) => {
-            this.linterMessages = this.linterMessages.concat(messages);
-          });
-
-          resolve({
-            linterMessages: this.linterMessages,
-            scannedFiles: [this.filename],
-          });
-        })
-        .catch(reject);
+  async scan(_rules = this._defaultRules) {
+    const contents = await this.getContents();
+    // Ignore private functions exported in rule files.
+    //
+    // (These are exported for testing purposes, but we don't want
+    // to include them in our linter's rules.)
+    const rules = ignorePrivateFunctions(_rules);
+
+    const ruleResults = await Promise.all(Object.keys(rules).map((rule) => {
+      this._rulesProcessed++;
+      return rules[rule](contents, this.filename, this.options);
+    }));
+
+    ruleResults.forEach((messages) => {
+      this.linterMessages = this.linterMessages.concat(messages);
     });
-  }
 
-  getContents() {
-    return new Promise((resolve, reject) => {
-      if (this._parsedContent !== null) {
-        return resolve(this._parsedContent);
-      }
-
-      return this._getContents()
-        .then((contents) => {
-          this._parsedContent = contents;
+    return {
+      linterMessages: this.linterMessages,
+      scannedFiles: [this.filename],
+    };
+  }
 
-          resolve(this._parsedContent);
-        })
-        .catch(reject);
-    });
+  async getContents() {
+    if (this._parsedContent !== null) {
+      return this._parsedContent;
+    }
+    this._parsedContent = await this._getContents();
```

Comment:
I would defer it for a follow up, because it is sightly different from what this function was doing before the async/await refactoring, but to prevent `this._getContents` to be called twice if `getContents` has been called twice while the this._getContents() is still not resolved, it could be better to store the promise and return it from the async function, e.g.:

```js
async getContents() {
  if (this._promiseParsedContent !== null)
    return this._promiseParsedContent;
  }

  this._promiseParsedContent = this._getContents();

  return this._promiseParsedContent
}
```